### PR TITLE
Update ServerCapabilities structure

### PR DESCRIPTION
**Pull request by mniewrzal**
Added fields: documentLinkProvider, executeCommandProvider, experimental

Signed-off-by: Michal Niewrzal <michal.n@zend.com>

### CODE REVIEW

**Comment by spoenemann** [comment]
We'll work on a 0.1.1 release this week. We should merge this once we've switched master to 0.2.0-SNAPSHOT.

**Comment by spoenemann** [comment]
Fixes #37.

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend [hunk 1]
```diff
@@ -817,6 +817,21 @@ class ServerCapabilities {
...
+	
+	/**
+	 * The server provides execute command support.
+	 */
+	ExecuteCommandOptions executeCommandProvider
+	
+	/**
+	 * Experimental server capabilities.
+	 */
+	Object experimental
```
Comment (spoenemann) [inline org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend]: I see the `experimental` field only for the ClientCapabilities?
https://github.com/Microsoft/language-server-protocol/blob/dbaeumer/3.0-spec/protocol.md

**Comment by akosyakov** [comment]
Closed in favour of https://github.com/eclipse/lsp4j/pull/61

**Pull request closed without merging**